smart/model_assembly.py

```diff
@@ -51,6 +53,16 @@
 # ====================================================
 
 
+class ParameterType(Enum):
+    """
+    The different types of parameter input
+    """
+
+    from_file = 1
+    constant = 2
+    expression = 3
+
```

Comment:
Another alternative would be to use
```python
class ParameterType(str, Enum):
    from_file = "from_file"
    constant = "constant"
    expression = "expression"
```
That way the user could also pass a string as an argument

Reply:
But do we really need that?:P

Reply:
I think the API would be simpler if we can allow strings as well